client/extensions/woocommerce/woocommerce-services/views/carrier-accounts/dynamic-settings-form.js

```diff
@@ -0,0 +1,199 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import CompactCard from 'components/card/compact';
+import Checkbox from 'woocommerce/woocommerce-services/components/checkbox';
+import TextField from 'woocommerce/woocommerce-services/components/text-field';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import * as api from 'woocommerce/woocommerce-services/api';
+import { errorNotice, successNotice } from 'state/notices/actions';
+
+export const CheckboxFormFieldSet = ( props ) => {
+    return (
+        <>
+            <Checkbox
+                id={ props.fieldKey }
+                onChange={ props.onChange }
+                checked={ props.checked || false }
+            />
+            <label htmlFor={ props.fieldKey }><span>{ props.labelName }</span></label>
+        </>
+    );
+};
+
+export const DynamicCarrierAccountSettingsForm = ( props ) => {
+	const {
+        translate,
+    } = props;
+
+    const listOfFormFieldSet = [];
+
+    const [formFields, setFormFields] = useState({});
+    const [isSaving, setIsSaving] = useState(false);
+
+    const updateValue = ( fieldKey ) => ( newValue ) => {
+        const formFieldsCopy = {...formFields};
+        formFieldsCopy[fieldKey] = newValue;
+        setFormFields(formFieldsCopy);
+    }
+
+    const toggleCheckbox = ( fieldKey ) => () => {
+        const formFieldsCopy = {...formFields};
+        formFieldsCopy[fieldKey] = !formFieldsCopy[fieldKey];
+        setFormFields(formFieldsCopy);
+    };
+
+    const getInputComponent = (fieldKey) => {
```

Comment:
What do you think about moving the `getInputComponent` to a component factory? This allows React to better manage the component tree and its updates, without having to re-generate the components each time.